### DISCUSSION TITLE
Visual Studio 2022 Mac does not stop at breakpoints

### DISCUSSION THREAD

**Discussion by marceloarteaga** [aws/aws-lambda-dotnet#1254] [Q&A] [answered]
Using Visual Studio 2022 Mac
When setting up debugging in a project for a lambda function. The tool is executed showing pgae in http://localhost:5050/, but does not stop at breakpoints.

![image](https://user-images.githubusercontent.com/93331900/180631262-0b7e1eda-80d2-4822-81ff-ffa294122f90.png)

Project Configuration used:

**External Program**
/Users/USERNAME/.dotnet/tools/.store/amazon.lambda.testtool-6.0/0.12.3/amazon.lambda.testtool-6.0/0.12.3/tools/net6.0/any/Amazon.Lambda.TestTool.BlazorTester.dll

**Working Directory**
/Users/USERNAME/SOLUTIONDIRECTORY/PROJECTDIRECTORY/bin/Debug/net6.0

**Terminal Output**
```
AWS .NET Core 6.0 Mock Lambda Test Tool (0.12.3)
Loaded local Lambda runtime from project output /Users/USERNAME/SOLUTIONDIRECTORY/PROJECTDIRECTORY/src/PROJECTDIRECTORY/bin/Debug/net6.0
Found Lambda config file /Users/USERNAME/SOLUTIONDIRECTORY/PROJECTDIRECTORY/src/PROJECTDIRECTORY/aws-lambda-tools-defaults.json
Environment running at http://localhost:5050
```

Any observation that would be useful to make the debugging stop at breakpoints?

**Comment by ashishdhingra** [comment]
@marceloarteaga I was able to reproduce the issue where breakpoints are not hit when using Visual Studio 2022 for Mac. Using the same project in Jetbrains Rider hits the breakpoint successfully with test tool.

**Reproduction steps:**
1. Use command `dotnet tool update -g Amazon.Lambda.TestTool-6.0` to install latest version of test tool (currently `0.12.3`)
2. Use command `dotnet new -i "Amazon.Lambda.Templates::*"` to install latest templates.
3. Use command `dotnet new lambda.EmptyFunction --name BlogFunction --profile default --region us-east-2` to create a simple lambda function project.
4. Open project using Visual Studio 2022 for Mac.
5. Follow steps at https://github.com/aws/aws-lambda-dotnet/tree/master/Tools/LambdaTestTool#configure-for-visual-studio-for-mac to configure Lambda Test tool for Mac (use path to `Amazon.Lambda.TestTool.BlazorTester.dll`, e.g., `/Users/<<username>>/.dotnet/tools/.store/amazon.lambda.testtool-6.0/0.12.3/amazon.lambda.testtool-6.0/0.12.3/tools/net6.0/any/Amazon.Lambda.TestTool.BlazorTester.dll)
6. Set the breakpoint.
7. Debug the project.

@normj Please advise if some additional step is required for Visual Studio 2022 for Mac.

**Comment by normj** [comment]
In the test tool when you click the `Execute Function` button are you getting any output? Does it look like the the function ran but just didn't stop at the breakpoint? If you could add a screenshot of the settings currently set in the test tool that would be great.

**Reply by ashishdhingra** [reply]
@normj Here are the required inputs:
- When I click Execute function, we do get the output.
- Yes, the function ran, but didn't stop at the breakpoint.
- Below is the screenshot.
<img width="2286" alt="Screen Shot 2022-08-23 at 9 34 51 AM" src="https://user-images.githubusercontent.com/67916761/186213164-9d5eab39-3463-4f86-9a3d-eae9bddc855d.png">

**Reply by marceartist** [reply]
![Capture 2022-08-23 at 16 43 04](https://user-images.githubusercontent.com/7375067/186272216-8e841348-96b4-4c2f-9690-77b033e30fd9.png)

Screenshot of the project configuration

**Comment by marceartist** [comment, marked as answer]
I got the breakpoints to work by attaching the **AWS .NET Core 6.0 Mock Lambda Test Tool** dll to the process:
```
dotnet Amazon.Lambda.TestTool.BlazorTester.dll.
```

Steps:
1. Configure the Project to use **AWS .NET Core 6.0 Mock Lambda Test Tool**.
2. Place a breakpoint
3. Debug project
4. Attach the **AWS .NET Core 6.0 Mock Lambda Test Tool** dll to the process. Menu: **Debug**, option: **Attach to Process...**
5. Press the button: **Execute function**.

![image](https://user-images.githubusercontent.com/7375067/186275325-fc4e64bf-5e5d-4906-8a3d-134ef6331b13.png)

The process must stop at the breakpoint:
![image](https://user-images.githubusercontent.com/7375067/186274970-ea801871-a808-4ae7-898f-fb93bea18303.png)

**Reply by derekvantonder3DD** [reply]
Bro, this saved my life today.
